Plugins/NativeXr/Source/NativeXr.cpp

```diff
@@ -2572,7 +2572,9 @@ namespace Babylon
 
                         auto napiResultRef{Napi::Weak(napiResult)};
                         auto imageTrackingArray{m_imageTrackingResultsArray.Value()};
-                        imageTrackingArray.Set(imageTrackingArray.Length(), napiResultRef.Value());
+                        auto index = imageTrackingArray.Length();
+                        imageTrackingArray.Set("length", Napi::Value::From(env, index + 1));
```

Comment:
Should this be handled by NAPI implementation?

Reply:
Yes probably as this is an issue only in Hermes but I'm not sure if the issue is in NAPI or Hermes.  I'm not sure if I want to scope that to be part of this change though.

Reply:
No, probably not. But we can try to fix it separately beforehand maybe?

Reply:
Managed to get this working, and opened another PR here:
https://github.com/BabylonJS/BabylonNative/pull/1037